x/distribution/abci.go

```
@@ -23,11 +23,17 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper) error {
 	// TODO this is Tendermint-dependent
 	// ref https://github.com/cosmos/cosmos-sdk/issues/3095
 	if ctx.BlockHeight() > 1 {
-		k.AllocateTokens(ctx, previousTotalPower, ctx.VoteInfos())
+		err := k.AllocateTokens(ctx, previousTotalPower, ctx.VoteInfos())
+		if err != nil {
+			panic(err)
```

let's return an error instead.